model/client4.go

```diff
@@ -3962,287 +3943,287 @@ func (c *Client4) UploadLicenseFile(data []byte) (bool, *Response) {
+	if err != nil {
+		return false, BuildResponse(r), err
 	}
 	defer closeBody(r)
-	return true, BuildResponse(r)
+	return true, BuildResponse(r), nil
 }
 
 // DeletePreferences deletes the user's preferences.
-func (c *Client4) DeletePreferences(userId string, preferences *Preferences) (bool, *Response) {
+func (c *Client4) DeletePreferences(userId string, preferences *Preferences) (bool, *Response, error) {
 	buf, err := json.Marshal(preferences)
 	if err != nil {
-		return false, BuildErrorResponse(nil, NewAppError("DeletePreferences", "api.marshal_error", nil, err.Error(), http.StatusInternalServerError))
+		return false, nil, NewAppError("DeletePreferences", "api.marshal_error", nil, err.Error(), http.StatusInternalServerError)
 	}
-	r, appErr := c.doApiPostBytes(c.GetPreferencesRoute(userId)+"/delete", buf)
-	if appErr != nil {
-		return false, BuildErrorResponse(r, appErr)
+	r, err := c.doApiPostBytes(c.GetPreferencesRoute(userId)+"/delete", buf)
+	if err != nil {
+		return false, BuildResponse(r), err
 	}
 	defer closeBody(r)
-	return true, BuildResponse(r)
+	return true, BuildResponse(r), nil
 }
 
 // GetPreferencesByCategory returns the user's preferences from the provided category string.
-func (c *Client4) GetPreferencesByCategory(userId string, category string) (Preferences, *Response) {
+func (c *Client4) GetPreferencesByCategory(userId string, category string) (Preferences, *Response, error) {
 	url := fmt.Sprintf(c.GetPreferencesRoute(userId)+"/%s", category)
 	r, err := c.DoApiGet(url, "")
 	if err != nil {
-		return nil, BuildErrorResponse(r, err)
+		return nil, BuildResponse(r), err
 	}
 	defer closeBody(r)
 	preferences, _ := PreferencesFromJson(r.Body)
-	return preferences, BuildResponse(r)
+	return preferences, BuildResponse(r), nil
 }
 
 // GetPreferenceByCategoryAndName returns the user's preferences from the provided category and preference name string.
-func (c *Client4) GetPreferenceByCategoryAndName(userId string, category string, preferenceName string) (*Preference, *Response) {
+func (c *Client4) GetPreferenceByCategoryAndName(userId string, category string, preferenceName string) (*Preference, *Response, error) {
 	url := fmt.Sprintf(c.GetPreferencesRoute(userId)+"/%s/name/%v", category, preferenceName)
 	r, err := c.DoApiGet(url, "")
 	if err != nil {
-		return nil, BuildErrorResponse(r, err)
+		return nil, BuildResponse(r), err
 	}
 	defer closeBody(r)
-	return PreferenceFromJson(r.Body), BuildResponse(r)
+	return PreferenceFromJson(r.Body), BuildResponse(r), nil
 }
 
 // SAML Section
 
 // GetSamlMetadata returns metadata for the SAML configuration.
-func (c *Client4) GetSamlMetadata() (string, *Response) {
+func (c *Client4) GetSamlMetadata() (string, *Response, error) {
 	r, err := c.DoApiGet(c.GetSamlRoute()+"/metadata", "")
 	if err != nil {
-		return "", BuildErrorResponse(r, err)
+		return "", BuildResponse(r), err
 	}
 	defer closeBody(r)
 	buf := new(bytes.Buffer)
 	_, _ = buf.ReadFrom(r.Body)
```

Comment:
Unrelated, but will be good to handle the error here.

Reply:
Good suggestion. Added in 3ac313456.